library/cwm/src/lib/cwm/page.rb

```diff
@@ -0,0 +1,53 @@
+module CWM
+  # A page widget is a group of widgets that are contained within it.
+  # Several pages are in turn contained within a {Pager}.
+  #
+  # {TreePager} is a {Pager}.
+  #
+  # {Tabs} is a {Pager} and a {Tab} is its {Page}
+  # (FIXME: We haven't made Tabs a subclass of Pager yet.
+  # That will come after we have TreePager working.
+  class Page < CustomWidget
+    # @return [Boolean] is this the initially selected tab
+    attr_accessor :initial
+
+    # @return [WidgetTerm] contents of the tab, can contain {AbstractWidget}s
+    attr_reader :contents
+    # @return [String] label defines name of tab header
+    attr_reader :label
+    attr_reader :widget_id
```

Comment:
It is fine for me, but `CustomWidget` already have widget_id, so it should work without it. And if you need to assign widget_id. You can do it in initialize like `self.widget_id = widget_id` even if it is still against idea of widget_id. Which is by default taken from class name. So if you have `class DisksPage < Page` it will have widget_id "DisksPage" and if it have multiple instance in one UI, then it can set it on its own.